janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/JanusGraphStep.java

```diff
@@ -78,7 +80,14 @@ public String toString() {
 
     @Override
     public void addAll(Iterable<HasContainer> has) {
-        Iterables.addAll(hasContainers, has);
+        has.forEach(hasContainer -> {
```

Comment:
Logic here and in JanusGraphVertexStep looks quite similar. Can we refactor, reuse the logic, and test it someplace?